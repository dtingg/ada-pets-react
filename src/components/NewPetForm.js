import React, { Component } from 'react';
import PropTypes from 'prop-types';

import 'bootstrap/dist/css/bootstrap.min.css';
import './NewPetForm.css'

class NewPetForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: "",
      species: "",
      location: "",
      image: "",
      about: "",
    };
  }
  
  onNameChange = (event) => {
    console.log("Name field updated");
    this.setState({
      name: event.target.value,
    });
  }

  onSpeciesChange = (event) => {
    console.log("Species field updated");
    this.setState({
      name: event.target.value,
    });
  }

  onLocationChange = (event) => {
    console.log("Location field updated");
    this.setState({
      location: event.target.value,
    });
  }

  onImageChange = (event) => {
    console.log("Image field updated");
    this.setState({
      image: event.target.value,
    });
  }

  onAboutChange = (event) => {
    console.log("About field updated");
    this.setState({
      about: event.target.value,
    });
  }

  onFormSubmit = (event) => {
    event.preventDefault();
  
    const newPet = {
      name: this.state.name,
      species: this.state.species,
      location: this.state.location,
      image: this.state.image,
      about: this.state.about,
    };

    this.setState({
      name: "",
      species: "",
      location: "",
      image: "",
      about: "",
    });

    console.log(newPet)

    // this.props.addPetCallback(newPet);

  }

  render() {
    return (
      <form  className="new-pet-form" onSubmit={this.onFormSubmit}>
        <h3>Add a Pet</h3>
        { /* A form should go here! */ }
        <div>
          <label htmlFor="name">Name:</label>
          <input onChange={this.onNameChange} value={this.state.name} name="name" />
        </div>







        <input className="btn btn-success new-pet-form--submit" type="submit" name="submit" value="Add a Pet" />
      </form>
    );
  }


}

NewPetForm.propTypes = {
  addPetCallback: PropTypes.func.isRequired,
};

export default NewPetForm;