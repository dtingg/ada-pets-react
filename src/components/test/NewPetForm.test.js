import React from "react";
import { render, cleanup } from "@testing-library/react"
import NewPetForm from "../NewPetForm";


