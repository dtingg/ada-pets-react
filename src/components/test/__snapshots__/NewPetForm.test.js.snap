// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPetForm Snapshot test for NewPetForm 1`] = `
<DocumentFragment>
  <form
    class="new-pet-form"
  >
    <h3>
      Add a Pet
    </h3>
    <div>
      <label
        class="new-pet-form--label"
        for="name"
      >
        Name
      </label>
      <input
        name="name"
        placeholder="name"
        value=""
      />
    </div>
    <div>
      <label
        class="new-pet-form--label"
        for="name"
      >
        Species
      </label>
      <input
        placeholder="species"
        species="species"
        value=""
      />
    </div>
    <div>
      <label
        class="new-pet-form--label"
        for="name"
      >
        Location
      </label>
      <input
        location="location"
        placeholder="location"
        value=""
      />
    </div>
    <div>
      <label
        class="new-pet-form--label"
        for="name"
      >
        Image
      </label>
      <input
        image="image"
        placeholder="image link"
        value=""
      />
    </div>
    <div>
      <label
        class="new-pet-form--label"
        for="name"
      >
        About
      </label>
      <div>
        <textarea
          about="about"
          class="new-pet-form--about"
        />
      </div>
    </div>
    <input
      class="btn btn-success new-pet-form--submit"
      name="submit"
      type="submit"
      value="Add a Pet"
    />
  </form>
</DocumentFragment>
`;
